core/trie.go

```diff
@@ -144,6 +146,21 @@ func NewTrie(storage TrieStorage, height uint) *Trie {
 	}
 }
 
+// DoOnTempTrie creates an in-memory Trie of height `height` and runs `do` on that Trie
+func DoOnTempTrie(height uint, do func(*Trie) error) error {
```

Comment:
Also can we please rename this to `RunOnTempTrie`?

Reply:
And should be private?

Reply:
I actually chose `DoOnTempTrie` over `RunOnTempTrie`. But it was so close, will change:)

>  And should be private?

nope it will be used in commitments calculation.